### ISSUE TITLE
Feels fast, but lots of ANRs

### ISSUE THREAD

**Issue by CorruptedPlazma** [cb-linux/breath#204]
After some trial and error I managed to get Ubuntu loaded onto a flash drive and booted, I started with setting up Gnome (I opted for a clean start with just CLI, and installing what I wanted from there), rebooted, and Gnome runs fine, better than fine, it's very fast. However, applications seem to freeze up and receive the "app not responding" error. Usually the apps will recover, other times it can lead to the entire system freezing up. System lockups seem to be alleviated with the addition of swap space, however. 

tl;dr: Feels fast, but freezes often. 

For reference, I'm using an HP Chromebook 14a, Blooglet (octopus) board. 
Model: BLOOGLET C2B-A4G-D2K-S7R-M4Q-A3R

**Comment by MilkyDeveloper** [comment]
Swap isn't supported within the ChromeOS kernel that this project is using. Just curious, how did you modify the swap?

ChromeOS's substitute for swap is `zram`. Try running this from ze beloved [Gentoo Wiki](https://wiki.gentoo.org/wiki/Zram):
```
modprobe zram

SIZE=6144
echo $(($SIZE*1024*1024)) > /sys/block/zram0/disksize

mkswap /dev/zram0

swapon /dev/zram0 -p 10
```

This creates a "partition" in your ram that holds 6GB, but is actually compressed to 2GB. If you're on a system with 4 gigs of ram, this creates a system with 10GB of ram.

**Comment by CorruptedPlazma** [comment]
Must be a placebo then, as I just created a partition of swap space on the drive I'm using (most of my experience with Linux comes from Arch, and I completely forgot ChromeOS uses zram over swap). I'll try setting up zram and see if that does anything to help.

**Comment by CorruptedPlazma** [comment]
After configuring zram, there appears to be some small improvements, system still appears to be locking up even as I type this.

**Comment by MilkyDeveloper** [comment]
Oops, I forgot, it’s probably because you’re running of of a USB. Ubuntu uncompresses their live USB to RAM, but this is full-fledged Ubuntu. Try using a faster USB 3.1/USB C drive or installing to internal storage.

**Comment by CorruptedPlazma** [comment]
I would definitely like to continue going off a USB drive, since I plan to grab a 256GB mini drive to run Linux off of, so I don't have to deal with the measly 32GBs of emmc storage inside of my device. I was beginning to think the problem was my storage as well, I'm finding it quite challenging to find any specs of the USB port on my device, and I'm using a cheap USB 3.0 drive from a not-so-reputable brand. Building f2fs may also serve a lot of use for plenty of others too since there are quite a few Chromebooks that are internet oriented and as such have lacking storage (I've seen machines with 16GBs in them - I can't possibly imagine a full, modern OS with a desktop environment would be comfortable on that.)

**Comment by MilkyDeveloper** [comment]
I think your only solution is to get a better USB drive that has USB 3.1. f2fs's improvements are minimal at best. This same error would apply to any other OS booted of a USB.
